pkg/generators/union.go

```diff
@@ -163,7 +163,7 @@ func parseUnionStruct(t *types.Type) (*union, []error) {
 		if types.ExtractCommentTags("+", m.CommentLines)[tagUnionDiscriminator] != nil {
 			errors = append(errors, u.setDiscriminator(jsonName)...)
 		} else {
-			if !hasOptionalTag(&m) {
+			if optional, err := hasOptionalTag(&m); !optional || err != nil {
```

Comment:
Do we have users for this code path? Should the error be passed forward?

Reply:
The error should already be logged by the other code paths

Reply:
sounds good